src/models/ServerAcl.ts

```diff
@@ -32,6 +32,30 @@ export class ServerAcl {
 
     }
 
+    /**
+     * Initialize the ServerACL with the rules from an existing ACL object retrieved from the room state.
+     * @param acl The content of a `m.room.server_acl` event.
+     */
+    public fromACL(acl: any): ServerAcl {
+        if (this.allowedServers.size !== 0 || this.deniedServers.size !== 0) {
+            throw new TypeError(`Expected this ACL to be uninitialized.`);
+        }
+        const allow = acl['allow'];
+        const deny = acl['deny'];
+        const ips = acl['allow_ip_literals'];
+
+        if (Array.isArray(allow)) {
+            allow.forEach(this.allowServer, this);
```

Comment:
If we don't trust `allow`, we should validate that each value in this array is a string. Same for `deny`.